src/alignment.py

```diff
@@ -0,0 +1,147 @@
+        # Fill out first row
+        for j in range(0, n + 1):
+            score[0][j] = gap_penalty * j
+        
+        # Fill out all other values in the score matrix
+        for i in range(1, m + 1):
+            for j in range(1, n + 1):
+                # Calculate the score by checking the top, left, and diagonal cells
+                match = score[i - 1][j - 1] + match_score(seq1[j-1], seq2[i-1])
+                delete = score[i - 1][j] + gap_penalty
+                insert = score[i][j - 1] + gap_penalty
+                # Record the maximum score from the three possible scores calculated above
+                score[i][j] = max(match, delete, insert)
+        
+        ## Traceback and compute the alignment 
+        
+        # Create variables to store alignment
+        align1 = ""
+        align2 = ""
+        
+        # Start from the bottom right cell in matrix
+        i = m
+        j = n
+        
+        # We'll use i and j to keep track of where we are in the matrix, just like above
+        while i > 0 and j > 0: # end touching the top or the left edge
+            score_current = score[i][j]
+            score_diagonal = score[i-1][j-1]
+            score_up = score[i][j-1]
+            score_left = score[i-1][j]
+            
+            # Check to figure out which cell the current score was calculated from,
+            # then update i and j to correspond to that cell.
+            if score_current == score_diagonal + match_score(seq1[j-1], seq2[i-1]):
+                align1 += seq1[j-1]
+                align2 += seq2[i-1]
+                i -= 1
+                j -= 1
+            elif score_current == score_up + gap_penalty:
+                align1 += seq1[j-1]
+                align2 += '-'
+                j -= 1
+            elif score_current == score_left + gap_penalty:
+                align1 += '-'
+                align2 += seq2[i-1]
+                i -= 1
+
+        # Finish tracing up to the top left cell
+        while j > 0:
+            align1 += seq1[j-1]
+            align2 += '-'
+            j -= 1
+        while i > 0:
+            align1 += '-'
+            align2 += seq2[i-1]
+            i -= 1
+        
+        # Since we traversed the score matrix from the bottom right, our two sequences will be reversed.
+        # These two lines reverse the order of the characters in each sequence.
+        align1 = align1[::-1]
+        align2 = align2[::-1]
+        
+        return align2
+
+    def align(self):
+        
+        align_scores={}
+        for i in range(len(self.list_of_seq)):
+            keys = [i for i in range(len(self.list_of_seq))]
+            keys.remove(i)
+            values =[]
+            for j in keys:
+                values.append(self.wunsch(str(self.list_of_seq[i]), str(self.list_of_seq[j])))
+                j+=1
+            res = dict(zip(keys,values))
+            align_scores[i] = res
+
+        return align_scores
+
+if __name__ == "__main__":
```

Comment:
Chcę żeby podany kod zadziałał:

```
    al = Alignment('TCgA', 'CCAAGT', 'AttTGC', 'TTTGGCTG')
    print(str(al))
    print(al)
    print(len(al))


    print(al[2]) # To zwróci 3 sekwencję  
    print(al[2,0]) # a to zwróci aligmnet 3 z 1 sekwencją

```

Reply:
Pamiętaj o funkcji __repr__ w seqence żeby Ci się ładnie wyświetlało w listach i słownikach

Reply:
Można dodać coś w stylu, że to sekwencja:
![image](https://user-images.githubusercontent.com/40667237/154488931-d935d406-7678-4ef8-8c0d-b0ece761500d.png)